internal/audit/audit.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package audit
+
+import (
+	"crypto/md5"
+	"fmt"
+	"github.com/juju/loggo"
+	"strings"
+)
+
+var logger = loggo.GetLogger("audit")
+
+
+type Operation string
+
+const (
+	SET_READ Operation = "set_read"
+	SET_WRITE Operation = "set_write"
+)
+
+func LogAcl(entity string, operation Operation, final_state []string) {
+	state := make([]string, len(final_state))
+	for k, v := range final_state {
+		hasher := md5.New()
```

Comment:
If the aim of this is to anonymise the audit log, I don't
think that this will do that job.

User names are not generally designed to be resistant
to dictionary attack. Even if we use a decent password
hash algorithm (e.g. pbkdf2) the number of possibilities
is generally low enough that it's unlikely to be much
help (plus that would make writing the log extremely
heavyweight in CPU time).

One possibility might be to store a secret key somewhere
and use that as the start of a cipher block chain
that encrypts all the names, but that only helps to protect
the audit log from random snooping - it doesn't
actually anonymise the names from us.

In general, I think the two goals (anonymisation and audit
log verification) are in opposition to one another - I'm
not convinced that it's possible to have an anonymous audit
log that is also verifiable against the actual database.

We should have a discussion about this, I think.